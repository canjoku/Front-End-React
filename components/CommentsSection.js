import { Component } from 'react'
import CommentsForm from './CommentsForm'
import CommentCard from './CommentCard'
import Auth from '../lib/Auth'

class CommentsSection extends Component {
  constructor(props){
    super(props)
    this.auth = new Auth()
    this.login = this.login.bind(this)
    this.logout = this.logout.bind(this)
  }

  login () {
    this.auth.login()
  }

  logout () {
    this.auth.logout()
    window.location.reload()
  }

  render () {
    const { comments, mutation, slug } = this.props
    if (this.auth.isAuthenticated()){
      return(
        <div>
          <CommentsForm mutation={mutation} slug={slug} />
          <a onClick={this.logout}>Logout</a>
          {comments.map((comment, index) => {
            return(
              <CommentCard key={index} comment={comment} />
            )
          })}
        </div>
      )
    } else {
      return(
        <div>
          <p>If you would like to leave a comment, then please login</p>
          <a onClick={this.login}>Login</a>
          {comments.map((comment, index) => {
            return(
              <CommentCard key={index} comment={comment} />
            )
          })}
        </div>
      )
    }
  }
}

export default CommentsSection


