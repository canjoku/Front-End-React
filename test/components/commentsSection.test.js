import CommentsSection from '../../components/CommentsSection'
jest.mock('../../lib/Auth.js')

test('<CommentsSection /> renders correctly', () => {
  const commentsArray = [
    {"body": "the first comment"},
    {"body": "the second comment"}
  ]
  const component = shallow(<CommentsSection comments={commentsArray} slug={"a-fantastic-post"} />)
  expect(component).toMatchSnapshot()
})
