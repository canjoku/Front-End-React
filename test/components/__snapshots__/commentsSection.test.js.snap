// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentsSection /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsSection
    comments={
      Array [
        Object {
          "body": "the first comment",
        },
        Object {
          "body": "the second comment",
        },
      ]
    }
    slug="a-fantastic-post"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          If you would like to leave a comment, then please login
        </p>,
        <a
          onClick={[Function]}
        >
          Login
        </a>,
        Array [
          <CommentCard
            comment={
              Object {
                "body": "the first comment",
              }
            }
          />,
          <CommentCard
            comment={
              Object {
                "body": "the second comment",
              }
            }
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you would like to leave a comment, then please login",
        },
        "ref": null,
        "rendered": "If you would like to leave a comment, then please login",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "comment": Object {
            "body": "the first comment",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "comment": Object {
            "body": "the second comment",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            If you would like to leave a comment, then please login
          </p>,
          <a
            onClick={[Function]}
          >
            Login
          </a>,
          Array [
            <CommentCard
              comment={
                Object {
                  "body": "the first comment",
                }
              }
            />,
            <CommentCard
              comment={
                Object {
                  "body": "the second comment",
                }
              }
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you would like to leave a comment, then please login",
          },
          "ref": null,
          "rendered": "If you would like to leave a comment, then please login",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "comment": Object {
              "body": "the first comment",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "comment": Object {
              "body": "the second comment",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
